src/ToricVarieties/NormalToricVarieties/constructors.jl

```diff
@@ -221,13 +262,13 @@ function toric_projective_space(d::Int)
     set_attribute!(variety, :dim, d)
     set_attribute!(variety, :dim_of_torusfactor, 0)
     set_attribute!(variety, :euler_characteristic, d+1)
-    set_attribute!(variety, :character_lattice, abelian_group([0 for i in 1:d]))
-    set_attribute!(variety, :torusinvariant_divisor_group, abelian_group([0 for i in 1:d+1]))
+    set_attribute!(variety, :character_lattice, free_abelian_group(d))
+    set_attribute!(variety, :torusinvariant_divisor_group, free_abelian_group(d+1))
     set_attribute!(variety, :map_from_cartier_divisor_group_to_torus_invariant_divisor_group, Hecke.identity_map(torusinvariant_divisor_group(variety)))
     set_attribute!(variety, :map_from_cartier_divisor_group_to_picard_group, map_from_weil_divisors_to_class_group(variety))
     set_attribute!(variety, :stanley_reisner_ideal, ideal([prod(Hecke.gens(cox_ring(variety)))]))
     set_attribute!(variety, :irrelevant_ideal, ideal(Hecke.gens(cox_ring(variety))))
-    betti_numbers = [if iseven(i) fmpz(1) else fmpz(0) end for i in 0:2*d]
+    betti_numbers = [fmpz(1) for i in 0:d]
```

Comment:
Could also do this (tiny bit faster / uses less memory -- but in reality it doesn't matter here, so pick what your taste prefers)
```suggestion
    betti_numbers = fill(fmpz(1), d+1)
```

Reply:
Thank you. Done in https://github.com/oscar-system/Oscar.jl/pull/940/commits/83fa4c7b9528f694fc8da99b125cbb15abe57319. (After the rebase, I could not commit your suggestion directly.)